### ISSUE TITLE
xcov issue - wrong number of arguments (0 for 1) (ArgumentError)

### ISSUE THREAD

**Issue by wqzhang123** [fastlane/fastlane#8407]
### New Issue Checklist

- [x ] Updated fastlane to the latest version
- [ x] I have read the [Contribution Guidelines](https://github.com/fastlane/fastlane/blob/master/CONTRIBUTING.md)

### Issue Description
/Users/a573655/.fastlane/bin/bundle/lib/ruby/gems/2.2.0/gems/xcov-1.1.0/lib/xcov/manager.rb:13:in `initialize': [!] wrong number of arguments (0 for 1) (ArgumentError)

##### Complete output when running fastlane, including the stack trace and command used
> You can use: `--capture_output` as the last commandline argument to get that collected for you

[09:08:38]: -------------------------------------------------
[09:08:38]: --- Step: Verifying required fastlane version ---
[09:08:38]: -------------------------------------------------
[09:08:38]: Your fastlane version 2.19.0 matches the minimum requirement of 2.5.0  ✅
[09:08:38]: ------------------------------
[09:08:38]: --- Step: default_platform ---
[09:08:38]: ------------------------------
[09:08:38]: Driving the lane 'ios test' 🚀
[09:08:38]: ------------------
[09:08:38]: --- Step: xcov ---
[09:08:38]: ------------------
+------------------+----------+
|        Lane Context         |
+------------------+----------+
| DEFAULT_PLATFORM | ios      |
| PLATFORM_NAME    | ios      |
| LANE_NAME        | ios test |
+------------------+----------+
[09:08:38]: wrong number of arguments (0 for 1)

+------+-------------------------------------+-------------+
|                     fastlane summary                     |
+------+-------------------------------------+-------------+
| Step | Action                              | Time (in s) |
+------+-------------------------------------+-------------+
| 1    | Verifying required fastlane version | 0           |
| 2    | default_platform                    | 0           |
| 💥   | xcov                                | 0           |
+------+-------------------------------------+-------------+

[09:08:38]: fastlane finished with errors

Looking for related GitHub issues on fastlane/fastlane...

/Users/a573655/.fastlane/bin/bundle/lib/ruby/gems/2.2.0/gems/xcov-1.1.0/lib/xcov/manager.rb:13:in `initialize': [!] wrong number of arguments (0 for 1) (ArgumentError)
	from /Users/a573655/.fastlane/bin/bundle/lib/ruby/gems/2.2.0/gems/fastlane-2.19.0/fastlane/lib/fastlane/actions/xcov.rb:8:in `new'
	from /Users/a573655/.fastlane/bin/bundle/lib/ruby/gems/2.2.0/gems/fastlane-2.19.0/fastlane/lib/fastlane/actions/xcov.rb:8:in `run'
	from /Users/a573655/.fastlane/bin/bundle/lib/ruby/gems/2.2.0/gems/fastlane-2.19.0/fastlane/lib/fastlane/runner.rb:243:in `block (2 levels) in execute_action'
	from /Users/a573655/.fastlane/bin/bundle/lib/ruby/gems/2.2.0/gems/fastlane-2.19.0/fastlane/lib/fastlane/actions/actions_helper.rb:50:in `execute_action'
	from /Users/a573655/.fastlane/bin/bundle/lib/ruby/gems/2.2.0/gems/fastlane-2.19.0/fastlane/lib/fastlane/runner.rb:221:in `block in execute_action'
	from /Users/a573655/.fastlane/bin/bundle/lib/ruby/gems/2.2.0/gems/fastlane-2.19.0/fastlane/lib/fastlane/runner.rb:217:in `chdir'
	from /Users/a573655/.fastlane/bin/bundle/lib/ruby/gems/2.2.0/gems/fastlane-2.19.0/fastlane/lib/fastlane/runner.rb:217:in `execute_action'
	from /Users/a573655/.fastlane/bin/bundle/lib/ruby/gems/2.2.0/gems/fastlane-2.19.0/fastlane/lib/fastlane/runner.rb:139:in `trigger_action_by_name'
	from /Users/a573655/.fastlane/bin/bundle/lib/ruby/gems/2.2.0/gems/fastlane-2.19.0/fastlane/lib/fastlane/fast_file.rb:146:in `method_missing'
	from Fastfile:35:in `block (2 levels) in parsing_binding'
	from /Users/a573655/.fastlane/bin/bundle/lib/ruby/gems/2.2.0/gems/fastlane-2.19.0/fastlane/lib/fastlane/lane.rb:33:in `call'
	from /Users/a573655/.fastlane/bin/bundle/lib/ruby/gems/2.2.0/gems/fastlane-2.19.0/fastlane/lib/fastlane/lane.rb:33:in `call'
	from /Users/a573655/.fastlane/bin/bundle/lib/ruby/gems/2.2.0/gems/fastlane-2.19.0/fastlane/lib/fastlane/runner.rb:49:in `block in execute'
	from /Users/a573655/.fastlane/bin/bundle/lib/ruby/gems/2.2.0/gems/fastlane-2.19.0/fastlane/lib/fastlane/runner.rb:45:in `chdir'
	from /Users/a573655/.fastlane/bin/bundle/lib/ruby/gems/2.2.0/gems/fastlane-2.19.0/fastlane/lib/fastlane/runner.rb:45:in `execute'
	from /Users/a573655/.fastlane/bin/bundle/lib/ruby/gems/2.2.0/gems/fastlane-2.19.0/fastlane/lib/fastlane/lane_manager.rb:52:in `cruise_lane'
	from /Users/a573655/.fastlane/bin/bundle/lib/ruby/gems/2.2.0/gems/fastlane-2.19.0/fastlane/lib/fastlane/command_line_handler.rb:30:in `handle'
	from /Users/a573655/.fastlane/bin/bundle/lib/ruby/gems/2.2.0/gems/fastlane-2.19.0/fastlane/lib/fastlane/commands_generator.rb:105:in `block (2 levels) in run'
	from /Users/a573655/.fastlane/bin/bundle/lib/ruby/gems/2.2.0/gems/commander-4.4.3/lib/commander/command.rb:178:in `call'
	from /Users/a573655/.fastlane/bin/bundle/lib/ruby/gems/2.2.0/gems/commander-4.4.3/lib/commander/command.rb:178:in `call'
	from /Users/a573655/.fastlane/bin/bundle/lib/ruby/gems/2.2.0/gems/commander-4.4.3/lib/commander/command.rb:153:in `run'
	from /Users/a573655/.fastlane/bin/bundle/lib/ruby/gems/2.2.0/gems/commander-4.4.3/lib/commander/runner.rb:446:in `run_active_command'
	from /Users/a573655/.fastlane/bin/bundle/lib/ruby/gems/2.2.0/gems/fastlane-2.19.0/fastlane_core/lib/fastlane_core/ui/fastlane_runner.rb:38:in `run!'
	from /Users/a573655/.fastlane/bin/bundle/lib/ruby/gems/2.2.0/gems/commander-4.4.3/lib/commander/delegates.rb:15:in `run!'
	from /Users/a573655/.fastlane/bin/bundle/lib/ruby/gems/2.2.0/gems/fastlane-2.19.0/fastlane/lib/fastlane/commands_generator.rb:302:in `run'
	from /Users/a573655/.fastlane/bin/bundle/lib/ruby/gems/2.2.0/gems/fastlane-2.19.0/fastlane/lib/fastlane/commands_generator.rb:42:in `start'
	from /Users/a573655/.fastlane/bin/bundle/lib/ruby/gems/2.2.0/gems/fastlane-2.19.0/fastlane/lib/fastlane/cli_tools_distributor.rb:65:in `take_off'
	from /Users/a573655/.fastlane/bin/bundle/lib/ruby/gems/2.2.0/gems/fastlane-2.19.0/bin/fastlane:15:in `<top (required)>'
	from /Users/a573655/.fastlane/bin/bundle/bin/fastlane:22:in `load'
	from /Users/a573655/.fastlane/bin/bundle/bin/fastlane:22:in `<main>'

### Environment

Please run `fastlane env` and copy the output below. This will help us help you :+1:
If you used `--capture_output` option please remove this block - as it is already included there.

[INSERT OUTPUT HERE]

**Comment by ohayon** [comment]
Hey @wqzhang123, could you please share the part of your Fastfile that calls `xcov`? Thanks! 🐙

**Comment by wqzhang123** [comment]
I have updated to 2.19.0, and xcov 1.1.1, still same issue, below is fastlane env
<details><summary>✅ fastlane environment ✅</summary>

### Stack

| Key                 | Value                                                         |
| ------------------- | ------------------------------------------------------------- |
| OS                  | 10.11.6                                                       |
| Ruby                | 2.2.4                                                         |
| Bundler?            | false                                                         |
| Git                 | git version 2.10.1 (Apple Git-78)                             |
| Installation Source | ~/.fastlane/bin/bundle/bin/fastlane                           |
| Host                | Mac OS X 10.11.6 (15G1217)                                    |
| Ruby Lib Dir        | ~/.fastlane/bin/bundle/lib                                    |
| OpenSSL Version     | OpenSSL 1.0.2g  1 Mar 2016                                    |
| Is contained        | true                                                          |
| Is homebrew         | false                                                         |
| Xcode Path          | /Applications/Xcodes/Xcode8.2.1/Xcode.app/Contents/Developer/ |
| Xcode Version       | 8.2.1                                                         |


### System Locale

| Variable | Value       |   |
| -------- | ----------- | - |
| LANG     | en_US.UTF-8 | ✅ |
| LC_ALL   | en_US.UTF-8 | ✅ |
| LANGUAGE | en_US.UTF-8 | ✅ |


### fastlane files:

<details><summary>`./fastlane/Fastfile`</summary>

```ruby
# Customise this file, documentation can be found here:
# https://github.com/fastlane/fastlane/tree/master/fastlane/docs
# All available actions: https://docs.fastlane.tools/actions
# can also be listed using the `fastlane actions` command

# Change the syntax highlighting to Ruby
# All lines starting with a # are ignored when running `fastlane`

# If you want to automatically update fastlane if a new version is available:
# update_fastlane

# This is the minimum version number required.
# Update this, if you use features of a newer version
fastlane_version "2.5.0"

default_platform :ios

platform :ios do
  before_all do
    # ENV["SLACK_URL"] = "https://hooks.slack.com/services/..."


  end

  desc "Runs all the tests"
  lane :test do

#swiftlint(
#        strict: true)

#    scan(
#        device: "iPhone 6"
#    )

    xcov(
        project: "./WFCore.xcodeproj",
        scheme: "WFCore",
        output_directory: "./fastlane/xcov_report"
    )

  end

  # You can define as many lanes as you want

  after_all do |lane|
    # This block is called, only if the executed lane was successful

    # slack(
    #   message: "Successfully deployed new App Update."
    # )

    jazzy(
        config: "fastlane/jazzy.config.yaml"
    )

  end

  error do |lane, exception|
    # slack(
    #   message: exception.message,
    #   success: false
    # )
  end
end


# More information about multiple platforms in fastlane: https://github.com/fastlane/fastlane/blob/master/fastlane/docs/Platforms.md
# All available actions: https://docs.fastlane.tools/actions

# fastlane reports which actions are used
# No personal data is recorded. Learn more at https://github.com/fastlane/enhancer
```
</details>

<details><summary>`./fastlane/Appfile`</summary>

```ruby
app_identifier "none" # The bundle identifier of your app
apple_id "" # Your Apple email address

team_id "[[DEV_PORTAL_TEAM_ID]]"  # Developer Portal Team ID

# you can even provide different app identifiers, Apple IDs and team names per lane:
# More information: https://github.com/fastlane/fastlane/blob/master/fastlane/docs/Appfile.md
```
</details>

### fastlane gems

| Gem      | Version | Update-Status  |
| -------- | ------- | -------------- |
| fastlane | 2.19.0  | 💥 Check failed |


### Loaded fastlane plugins:

**No plugins Loaded**


<details><summary><b>Loaded gems</b></summary>

| Gem                       | Version      |
| ------------------------- | ------------ |
| CFPropertyList            | 2.3.5        |
| activesupport             | 4.2.7.1      |
| addressable               | 2.5.0        |
| babosa                    | 1.0.2        |
| bigdecimal                | 1.2.6        |
| bundler                   | 1.14.3       |
| claide                    | 1.0.1        |
| colored                   | 1.2          |
| commander                 | 4.4.3        |
| domain_name               | 0.5.20161129 |
| dotenv                    | 2.2.0        |
| excon                     | 0.54.0       |
| faraday                   | 0.11.0       |
| faraday-cookie_jar        | 0.0.6        |
| faraday_middleware        | 0.11.0.1     |
| fastimage                 | 2.0.1        |
| gh_inspector              | 1.0.3        |
| google-api-client         | 0.9.26       |
| googleauth                | 0.5.1        |
| highline                  | 1.7.8        |
| http-cookie               | 1.0.3        |
| httpclient                | 2.8.3        |
| hurley                    | 0.2          |
| i18n                      | 0.7.0        |
| io-console                | 0.4.3        |
| json                      | 1.8.1        |
| jwt                       | 1.5.6        |
| libxml-ruby               | 2.9.0        |
| little-plugger            | 1.1.4        |
| logging                   | 2.1.0        |
| memoist                   | 0.15.0       |
| mime-types                | 3.1          |
| mime-types-data           | 3.2016.0521  |
| mini_magick               | 4.5.1        |
| minitest                  | 5.4.3        |
| multi_json                | 1.12.1       |
| multi_xml                 | 0.6.0        |
| multipart-post            | 2.0.0        |
| nanaimo                   | 0.2.3        |
| nokogiri                  | 1.5.6        |
| os                        | 0.9.6        |
| plist                     | 3.2.0        |
| power_assert              | 0.2.2        |
| psych                     | 2.0.8        |
| public_suffix             | 2.0.5        |
| rake                      | 10.4.2       |
| rdoc                      | 4.2.0        |
| representable             | 2.3.0        |
| retriable                 | 2.1.0        |
| rouge                     | 1.11.1       |
| rubygems-update           | 2.5.2        |
| rubyzip                   | 1.2.0        |
| security                  | 0.1.3        |
| signet                    | 0.7.3        |
| slack-notifier            | 1.5.1        |
| sqlite3                   | 1.3.13       |
| terminal-notifier         | 1.7.1        |
| terminal-table            | 1.7.3        |
| test-unit                 | 3.0.8        |
| thread_safe               | 0.3.5        |
| tty-screen                | 0.5.0        |
| tzinfo                    | 1.2.2        |
| uber                      | 0.0.15       |
| unf                       | 0.1.4        |
| unf_ext                   | 0.0.7.2      |
| unicode-display_width     | 1.1.3        |
| word_wrap                 | 1.0.0        |
| xcodeproj                 | 1.4.2        |
| xcov                      | 1.1.1        |
| xcpretty                  | 0.2.4        |
| xcpretty-travis-formatter | 0.0.4        |
</details>


*generated on:* **2017-03-01**
</details>

**Comment by wqzhang123** [comment]
my lane is very simple

desc "Runs all the tests"
  lane :test do

 swiftlint(
        strict: true)

  scan(
        device: "iPhone 6"
    )

    xcov(
        project: "./WFCore.xcodeproj",
        scheme: "WFCore",
        output_directory: "./fastlane/xcov_report"
    )

  end

**Comment by wqzhang123** [comment]
updated to fastlane 2.19.0 and xcov 1.1.1, same error. 

[11:44:45]: -------------------------------------------------
[11:44:45]: --- Step: Verifying required fastlane version ---
[11:44:45]: -------------------------------------------------
[11:44:45]: Your fastlane version 2.19.0 matches the minimum requirement of 2.5.0  ✅
[11:44:45]: ------------------------------
[11:44:45]: --- Step: default_platform ---
[11:44:45]: ------------------------------
[11:44:45]: Driving the lane 'ios test' 🚀
[11:44:45]: ------------------
[11:44:45]: --- Step: xcov ---
[11:44:45]: ------------------
+------------------+----------+
|        Lane Context         |
+------------------+----------+
| DEFAULT_PLATFORM | ios      |
| PLATFORM_NAME    | ios      |
| LANE_NAME        | ios test |
+------------------+----------+
[11:44:45]: wrong number of arguments (0 for 1)

+------+-------------------------------------+-------------+
|                     fastlane summary                     |
+------+-------------------------------------+-------------+
| Step | Action                              | Time (in s) |
+------+-------------------------------------+-------------+
| 1    | Verifying required fastlane version | 0           |
| 2    | default_platform                    | 0           |
| 💥   | xcov                                | 0           |
+------+-------------------------------------+-------------+

[11:44:45]: fastlane finished with errors

Looking for related GitHub issues on fastlane/fastlane...

/Users/a573655/.fastlane/bin/bundle/lib/ruby/gems/2.2.0/gems/xcov-1.1.1/lib/xcov/manager.rb:13:in `initialize': [!] wrong number of arguments (0 for 1) (ArgumentError)
	from /Users/a573655/.fastlane/bin/bundle/lib/ruby/gems/2.2.0/gems/fastlane-2.19.0/fastlane/lib/fastlane/actions/xcov.rb:8:in `new'
	from /Users/a573655/.fastlane/bin/bundle/lib/ruby/gems/2.2.0/gems/fastlane-2.19.0/fastlane/lib/fastlane/actions/xcov.rb:8:in `run'
	from /Users/a573655/.fastlane/bin/bundle/lib/ruby/gems/2.2.0/gems/fastlane-2.19.0/fastlane/lib/fastlane/runner.rb:243:in `block (2 levels) in execute_action'
	from /Users/a573655/.fastlane/bin/bundle/lib/ruby/gems/2.2.0/gems/fastlane-2.19.0/fastlane/lib/fastlane/actions/actions_helper.rb:50:in `execute_action'
	from /Users/a573655/.fastlane/bin/bundle/lib/ruby/gems/2.2.0/gems/fastlane-2.19.0/fastlane/lib/fastlane/runner.rb:221:in `block in execute_action'
	from /Users/a573655/.fastlane/bin/bundle/lib/ruby/gems/2.2.0/gems/fastlane-2.19.0/fastlane/lib/fastlane/runner.rb:217:in `chdir'
	from /Users/a573655/.fastlane/bin/bundle/lib/ruby/gems/2.2.0/gems/fastlane-2.19.0/fastlane/lib/fastlane/runner.rb:217:in `execute_action'
	from /Users/a573655/.fastlane/bin/bundle/lib/ruby/gems/2.2.0/gems/fastlane-2.19.0/fastlane/lib/fastlane/runner.rb:139:in `trigger_action_by_name'
	from /Users/a573655/.fastlane/bin/bundle/lib/ruby/gems/2.2.0/gems/fastlane-2.19.0/fastlane/lib/fastlane/fast_file.rb:146:in `method_missing'
	from Fastfile:35:in `block (2 levels) in parsing_binding'
	from /Users/a573655/.fastlane/bin/bundle/lib/ruby/gems/2.2.0/gems/fastlane-2.19.0/fastlane/lib/fastlane/lane.rb:33:in `call'
	from /Users/a573655/.fastlane/bin/bundle/lib/ruby/gems/2.2.0/gems/fastlane-2.19.0/fastlane/lib/fastlane/lane.rb:33:in `call'
	from /Users/a573655/.fastlane/bin/bundle/lib/ruby/gems/2.2.0/gems/fastlane-2.19.0/fastlane/lib/fastlane/runner.rb:49:in `block in execute'
	from /Users/a573655/.fastlane/bin/bundle/lib/ruby/gems/2.2.0/gems/fastlane-2.19.0/fastlane/lib/fastlane/runner.rb:45:in `chdir'
	from /Users/a573655/.fastlane/bin/bundle/lib/ruby/gems/2.2.0/gems/fastlane-2.19.0/fastlane/lib/fastlane/runner.rb:45:in `execute'
	from /Users/a573655/.fastlane/bin/bundle/lib/ruby/gems/2.2.0/gems/fastlane-2.19.0/fastlane/lib/fastlane/lane_manager.rb:52:in `cruise_lane'
	from /Users/a573655/.fastlane/bin/bundle/lib/ruby/gems/2.2.0/gems/fastlane-2.19.0/fastlane/lib/fastlane/command_line_handler.rb:30:in `handle'
	from /Users/a573655/.fastlane/bin/bundle/lib/ruby/gems/2.2.0/gems/fastlane-2.19.0/fastlane/lib/fastlane/commands_generator.rb:105:in `block (2 levels) in run'
	from /Users/a573655/.fastlane/bin/bundle/lib/ruby/gems/2.2.0/gems/commander-4.4.3/lib/commander/command.rb:178:in `call'
	from /Users/a573655/.fastlane/bin/bundle/lib/ruby/gems/2.2.0/gems/commander-4.4.3/lib/commander/command.rb:178:in `call'
	from /Users/a573655/.fastlane/bin/bundle/lib/ruby/gems/2.2.0/gems/commander-4.4.3/lib/commander/command.rb:153:in `run'
	from /Users/a573655/.fastlane/bin/bundle/lib/ruby/gems/2.2.0/gems/commander-4.4.3/lib/commander/runner.rb:446:in `run_active_command'
	from /Users/a573655/.fastlane/bin/bundle/lib/ruby/gems/2.2.0/gems/fastlane-2.19.0/fastlane_core/lib/fastlane_core/ui/fastlane_runner.rb:38:in `run!'
	from /Users/a573655/.fastlane/bin/bundle/lib/ruby/gems/2.2.0/gems/commander-4.4.3/lib/commander/delegates.rb:15:in `run!'
	from /Users/a573655/.fastlane/bin/bundle/lib/ruby/gems/2.2.0/gems/fastlane-2.19.0/fastlane/lib/fastlane/commands_generator.rb:302:in `run'
	from /Users/a573655/.fastlane/bin/bundle/lib/ruby/gems/2.2.0/gems/fastlane-2.19.0/fastlane/lib/fastlane/commands_generator.rb:42:in `start'
	from /Users/a573655/.fastlane/bin/bundle/lib/ruby/gems/2.2.0/gems/fastlane-2.19.0/fastlane/lib/fastlane/cli_tools_distributor.rb:65:in `take_off'
	from /Users/a573655/.fastlane/bin/bundle/lib/ruby/gems/2.2.0/gems/fastlane-2.19.0/bin/fastlane:15:in `<top (required)>'
	from /Users/a573655/.fastlane/bin/bundle/bin/fastlane:22:in `load'
	from /Users/a573655/.fastlane/bin/bundle/bin/fastlane:22:in `<main>'

**Comment by mattlangtree** [comment]
@wqzhang123 I was facing the same "wrong parameter count" issue and it seems that reverting to a lower version of xcov fixes this. In my case I locked my Gemfile to version "1.0.0" for xcov.

**Comment by wqzhang123** [comment]
@mattlangtree, yes, i was using xcov 1.0.1 only 2 days ago and worked fine, only after the update I'm seeing the issue

**Comment by wqzhang123** [comment]
@mattlangtree, nevermind, i figured out by using bundle Gemfile, but somehow i can't no longer install xcov 1.0.1, but i can install 1.0.0, and everything is working again with fastlane 2.19.1. I ope fastlane can fix the issue with the latest xcov.

**Comment by dodikk** [comment]
I've got ```Error parsing xcode file``` when I downgraded to `xcov 1.0.0` gem

**Comment by ohayon** [comment]
Hey @wqzhang123, instead of downgrading the version of _fastlane_ you are using, you could add `gem 'xcov', '1.0.0'` and then run `bundle install` to install it, and then when you want to execute _fastlane_ simply prefix the command with `bundle exec` 🚀 🐙

**Comment by wqzhang123** [comment]
@ohayon, thanks, that's exactly what I have done. I didn't downgrade fastlane, only xcov.
